nexus/social_media/services.py

```diff
@@ -0,0 +1,74 @@
+# Third Party Stuff
+import tweepy
+from django.conf import settings
+from django.utils import timezone
+
+# nexus Stuff
+from nexus.base import exceptions as exc
+from nexus.social_media.models import Post
+
+
+def update_post_object(post):
+    """Update post instance after the post is published.
+
+    :param post: Post model instance.
+
+    """
+    post.posted_time = timezone.now()
+    post.is_posted = True
+    post.save()
+
+
+def get_twitter_api_object(TWITTER_OAUTH):
+    """Function to generate a twitter api object.
+
+    :param TWITTER_OAUTH: A dictionary having essential twitter oauth tokens viz.
+    consumer_key, consumer_secret, access_key, access_secret.
+
+    :returns: twitter API object.
+
+    :raises WrongArguments: Exception, when invalid twitter oauth token(s) are provided.
+
+    """
+    try:
+        auth = tweepy.OAuthHandler(TWITTER_OAUTH['consumer_key'], TWITTER_OAUTH['consumer_secret'])
+        auth.set_access_token(TWITTER_OAUTH['access_key'], TWITTER_OAUTH['access_secret'])
+        twitter_api = tweepy.API(auth)
+        return twitter_api
+    except tweepy.error.TweepError:
+        raise exc.WrongArguments("TweepError: Invalid Twitter OAuth Token(s).")
+
+
+def post_to_twitter(post_id):
+    """Function to post on twitter.
+
+    :param post_id: UUID of the post instance to be posted.
+
+    :raises BadRequest: Exception, when unable to post to twitter using tweepy.
+
+    """
+    post = Post.objects.get(pk=post_id)
+    twitter_api = get_twitter_api_object(settings.TWITTER_OAUTH)
+
+    try:
+        if post.image:
+            filename = post.image.file.name
+            twitter_api.update_with_media(filename=filename, status=post.text, file=post.image)
```

Comment:
What happens if I provide the image as `None` here. Does this call post just status, or complain about no image?

Reply:
No, `this line` will not complain about `no image`. But as I said [here](https://github.com/CuriousLearner/nexus/pull/20#discussion_r263379993) the line above this `filename = post.image.file.name`, will raise an error.

Reply:
Then remove the if-else altogether, and for the filename, make a check.

Reply:
Okay :) I'll do that.